L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloPFClusterEmulator.h

```diff
@@ -0,0 +1,278 @@
+   gctEtaStrip_t s9;
+   gctEtaStrip_t s10;
+   gctEtaStrip_t s11;
+   gctEtaStrip_t s12;
+   gctEtaStrip_t s13;
+   gctEtaStrip_t s14;
+   gctEtaStrip_t s15;
+   gctEtaStrip_t s16;
+   gctEtaStrip_t s17;
+   gctEtaStrip_t s18;
+   gctEtaStrip_t s19;
+   gctEtaStrip_t s20;
+} region_t;
+
+GCTint_t bestOf2(const GCTint_t& t0, const GCTint_t& t1) {
+  GCTint_t x;
+  x = (t0.et > t1.et) ? t0 : t1;
+  return x;
+}
+
+GCTint_t getPeakOfStrip(const gctEtaStrip_t& etaStrip){
+  GCTint_t best12 = bestOf2(etaStrip.t[1],etaStrip.t[2]);
+  GCTint_t best34 = bestOf2(etaStrip.t[3],etaStrip.t[4]);
+  GCTint_t best56 = bestOf2(etaStrip.t[5],etaStrip.t[6]);
+  GCTint_t best1234 = bestOf2(best12, best34);
+  GCTint_t bestAll = bestOf2(best1234, best56);
+  
+  return bestAll;
+}
+
+GCTint_t getPeakBin(const gctEtaStripPeak_t& etaStripPeak){
+  GCTint_t best01 = bestOf2(etaStripPeak.p[0], etaStripPeak.p[1]);
+  GCTint_t best23 = bestOf2(etaStripPeak.p[2], etaStripPeak.p[3]);
+  GCTint_t best45 = bestOf2(etaStripPeak.p[4], etaStripPeak.p[5]);
+  GCTint_t best67 = bestOf2(etaStripPeak.p[6], etaStripPeak.p[7]);
+  GCTint_t best89 = bestOf2(etaStripPeak.p[8], etaStripPeak.p[9]);
+  GCTint_t best1011 = bestOf2(etaStripPeak.p[10], etaStripPeak.p[11]);
+  GCTint_t best1213 = bestOf2(etaStripPeak.p[12], etaStripPeak.p[13]);
+  GCTint_t best1415 = bestOf2(etaStripPeak.p[14], etaStripPeak.p[15]);
+  GCTint_t best1617 = bestOf2(etaStripPeak.p[16], etaStripPeak.p[17]);
+  GCTint_t best0123 = bestOf2(best01, best23);
+  GCTint_t best4567 = bestOf2(best45, best67);
+  GCTint_t best891011 = bestOf2(best89, best1011);
+  GCTint_t best12131415 = bestOf2(best1213, best1415);
+  GCTint_t best01234567 = bestOf2(best0123, best4567);
+  GCTint_t best01234567891011 = bestOf2(best01234567, best891011);
+  GCTint_t best121314151617 = bestOf2(best12131415, best1617);
+  GCTint_t best12131415161718 = bestOf2(best121314151617, etaStripPeak.p[18]);
+  GCTint_t bestAll = bestOf2(best01234567891011, best12131415161718);
+
+  return bestAll;
+}
+
+GCTint_t getPeakPosition(const region_t& region){
+  gctEtaStripPeak_t etaPeak;
+
+  etaPeak.p[0] = getPeakOfStrip(region.s1);
+  etaPeak.p[1] = getPeakOfStrip(region.s2);
+  etaPeak.p[2] = getPeakOfStrip(region.s3);
+  etaPeak.p[3] = getPeakOfStrip(region.s4);
+  etaPeak.p[4] = getPeakOfStrip(region.s5);
+  etaPeak.p[5] = getPeakOfStrip(region.s6);
+  etaPeak.p[6] = getPeakOfStrip(region.s7);
+  etaPeak.p[7] = getPeakOfStrip(region.s8);
+  etaPeak.p[8] = getPeakOfStrip(region.s9);
+  etaPeak.p[9] = getPeakOfStrip(region.s10);
+  etaPeak.p[10] = getPeakOfStrip(region.s11);
+  etaPeak.p[11] = getPeakOfStrip(region.s12);
+  etaPeak.p[12] = getPeakOfStrip(region.s13);
+  etaPeak.p[13] = getPeakOfStrip(region.s14);
+  etaPeak.p[14] = getPeakOfStrip(region.s15);
+  etaPeak.p[15] = getPeakOfStrip(region.s16);
+  etaPeak.p[16] = getPeakOfStrip(region.s17);
+  etaPeak.p[17] = getPeakOfStrip(region.s18);
+  etaPeak.p[18] = getPeakOfStrip(region.s19);
```

Comment:
Could any of this be condensed via a map and a loop? It may not make it simpler, so it isn't an issue.